src/EditorFeatures/Text/Extensions.SnapshotSourceText.cs

```diff
@@ -262,6 +239,53 @@ public override SourceText WithChanges(IEnumerable<TextChange> changes)
                     currentSnapshot: ((ITextSnapshot2)buffer.CurrentSnapshot).TextImage);
             }
 
+            private static ITextImage RecordReverseMapAndGetImage(ITextSnapshot editorSnapshot)
+            {
+                Contract.ThrowIfNull(editorSnapshot);
+
+                var textImage = ((ITextSnapshot2)editorSnapshot).TextImage;
+                Contract.ThrowIfNull(textImage);
+
+                // If we're already in the map, there's nothing to update.  Do a quick check
+                // to avoid two allocations per call to RecordTextSnapshotAndGetImage.
+                if (!s_roslynToEditorSnapshotMap.TryGetValue(textImage, out var weakReference))
```

Comment:
I recognize this got moved but it's still confusing: should this be called s_textImageToEditorSnapshotMap or something more precise?

Reply:
sure.